dependencyManagement/build.gradle.kts

```diff
@@ -21,12 +21,14 @@ plugins {
 
 data class DependencySet(val group: String, val version: String, val modules: List<String>)
 
+val TEST_SNAPSHOTS = rootProject.findProperty("testUpstreamSnapshots") == "true"
+
 val DEPENDENCY_BOMS = listOf(
   "com.fasterxml.jackson:jackson-bom:2.12.0",
   "com.google.protobuf:protobuf-bom:3.14.0",
   "com.linecorp.armeria:armeria-bom:1.3.0",
   "io.grpc:grpc-bom:1.34.0",
-  "io.opentelemetry:opentelemetry-bom:0.12.0",
+  "io.opentelemetry:opentelemetry-bom:${if (!TEST_SNAPSHOTS) "0.12.0" else "0.12.0"}",
```

Comment:
Should this point to SNAPSHOT version on the else case?

Reply:
Nah this needs to be the dependency that the upstream agent defines. It's annoying to maintain this, but it's probably ok (I might try a download POM and parse approach later).